modules/api/src/main/scala/cards/nine/api/NineCardsDirectives.scala

```diff
@@ -54,46 +53,59 @@ class NineCardsDirectives(
   with PathDirectives
   with RouteDirectives
   with SecurityDirectives
+  with FutureDirectives
   with JsonFormats {
 
   import cards.nine.api.accounts.JsonFormats._
 
-  implicit def fromTaskAuth[T](auth: ⇒ Task[Authentication[T]]): AuthMagnet[T] =
-    new AuthMagnet(onSuccess(auth))
+  implicit def fromTaskAuth[T](auth: ⇒ Task[AuthenticationResult[T]]): Future[AuthenticationResult[T]] =
+    auth.unsafePerformAsyncFuture()
+
+  val nineCardChallenge = HttpChallenge("NineCard", None)
 
   val rejectionByCredentialsRejected = AuthenticationFailedRejection(
-    cause            = AuthenticationFailedRejection.CredentialsRejected,
-    challengeHeaders = Nil
+    cause     = AuthenticationFailedRejection.CredentialsRejected,
+    challenge = nineCardChallenge
   )
 
+  def authenticate[A](result: Future[AuthenticationResult[A]]): Directive1[A] =
```

Comment:
Temporarily added to cover spray's `authenticate` directive. The built-in authentication directives of akka-http are more restrictive in the authentication method than of spray.